dj/api/nodes.py

```diff
@@ -446,6 +455,24 @@ def materialization_job_from_engine(engine: Engine) -> MaterializationJob:
     return engine_to_job_mapping[engine.dialect]  # type: ignore
 
 
+def filters_from_partitions(partitions: List[Partition]):
+    """
+    Derive filters needed from partitions spec.
+    """
+    filters = []
+    for partition in partitions:
+        if partition.type_ != PartitionType.TEMPORAL:
```

Comment:
Just realized that we probably need to include this filter for temporal partitions as well. Say, if we have the query:
```
SELECT dateint, foo FROM bar
```
then we probably need to add this filter when we materialize:
```
SELECT dateint, foo FROM bar WHERE dateint = {expression}
```
This should be sufficient to set make query right even if other tables need to reference the same (current) date value, as long as we join them to the partition column.

Reply:
Agreed that this needs to be done at some point. 

Do you think it's reasonable if this is done at the materialization/query service level, where it can add on the temporal filter after it also has the parameterized version of `dateint`? That's what I have set up at the moment. 

The reason I ask is that if we generate this with the filter appended, it won't be able to be parameterized based on the scheduled run parameters for `expression`. Or were you thinking that it would be a parameterized value that the materialization service can then replace? (I tried that at one point but DJ's AST won't allow for generating of templatized queries).

Reply:
> Do you think it's reasonable if this is done at the materialization/query service level, where it can add on the temporal filter after it also has the parameterized version of dateint?

Yes, I think we are on the same page. 

Wrt to materialization (or some generic query) parameters, I think we can try to get by w/o for the time being. Meaning that the special filters for temporal partitions will be tacked on during the materialization phase only. This would make the DJ queries cleaner imo. Ideally I'd like to stay away from some crazy SQL templating based on user notion of the "current" ot "execution" date. These concepts, if defined poorly, may lead to insane setup and misunderstanding of the run-date vs data-date. If we can handle that automatically it should be easier for the AE-users that DJ is targeted to.